### PR TITLE
251: Bypass PR update cache when the cooldown period expires for a bridge candidate

### PR DESCRIPTION
**Pull request by rwestberg**
Hi all,

Please review this change that ensures that the mailing list bridge bot retries the bridge operation if it was initially delayed due to the configured cooldown period.

Best regards,
Robin

## Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed

## Issue
[SKARA-251](https://bugs.openjdk.java.net/browse/SKARA-251): Bypass PR update cache when the cooldown period expires for a bridge candidate

## Approvers
 * Erik Helin ([ehelin](@edvbld) - **Reviewer**)

### CODE REVIEW

File: bots/mlbridge/src/main/java/org/openjdk/skara/bots/mlbridge/CooldownQuarantine.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    private String getUniqueId(PullRequest pr) {
+        var repo = pr.repository();
+        if (!repositoryIds.containsKey(repo)) {
+            repositoryIds.put(repo, Integer.toString(repositoryIds.size()));
+        }
+        return repositoryIds.get(repo) + ";" + pr.id();
+    }
+
+    public synchronized boolean inQuarantine(PullRequest pr) {
+        var uniqueId = getUniqueId(pr);
```
Comment (edvbld) [inline bots/mlbridge/src/main/java/org/openjdk/skara/bots/mlbridge/CooldownQuarantine.java]:
```suggestion
        var uniqueId = pr.webUrl();
```

File: bots/mlbridge/src/main/java/org/openjdk/skara/bots/mlbridge/CooldownQuarantine.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            log.info("Released from cooldown quarantine: " + pr.repository().name() + "#" + pr.id());
+            quarantineEnd.remove(uniqueId);
+            return false;
+        }
+        log.info("Quarantined due to cooldown: " + pr.repository().name() + "#" + pr.id());
+        return true;
+    }
+
+    public synchronized void updateQuarantineEnd(PullRequest pr, Instant end) {
+        var uniqueId = getUniqueId(pr);
```
Comment (edvbld) [inline bots/mlbridge/src/main/java/org/openjdk/skara/bots/mlbridge/CooldownQuarantine.java]:
```suggestion
        var uniqueId = pr.webUrl();
```

File: bots/mlbridge/src/main/java/org/openjdk/skara/bots/mlbridge/CooldownQuarantine.java (hunk 1 above)
Comment (rwestberg) [reply bots/mlbridge/src/main/java/org/openjdk/skara/bots/mlbridge/CooldownQuarantine.java]: Sounds reasonable!

**Pull request closed without merging**